test/integration/test_push_docker_image.py

```diff
@@ -0,0 +1,136 @@
+import contextlib
+import docker
+import json
+import logging
+import pytest
+import re
+import requests
+import time
+
+from docker.client import DockerClient
+from typing import Callable, Optional
+
+from test.ports import find_free_port
+from exasol.ds.sandbox.lib.dss_docker.push_image import DockerRegistry
+from exasol.ds.sandbox.lib.dss_docker import DssDockerImage
+
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.DEBUG)
+
+
+def normalize_request_name(name: str):
+    name = re.sub(r"[\[\]._]+", "_", name)
+    return re.sub(r"^_+|_+$", "", name)
+
+
+class LocalDockerRegistry(DockerRegistry):
```

Comment:
Currently class `LocalDockerRegistry` is only used in pytest tests. 
If we want to enable adding a cli option and test the cli against a registry based on a local Docker container from "registry:2" then we need to move this class to lib.

Reply:
We would need it in lib if we want to enable adding a cli option and test the cli against a registry based on a local Docker container from "registry:2".